lib/streamlit/state/widgets.py

```diff
@@ -108,26 +117,25 @@ def register_widget(
         dynamically-named functions.
     on_change_handler : Optional[WidgetCallback]
         An optional callback invoked when the widget's value changes.
-    deserializer : Optional[WidgetDeserializer]
-        Called to convert a widget's protobuf value to the value returned by
-        its st.<widget_name> function.
     args : Optional[WidgetArgs]
         args to pass to on_change_handler when invoked
     kwargs : Optional[WidgetKwargs]
         kwargs to pass to on_change_handler when invoked
 
     Returns
     -------
-    ui_value : Tuple[Any, bool]
+    ui_value : WidgetStateReport[T]
         - If our ScriptRunContext doesn't exist (meaning that we're running
-        a "bare script" outside of streamlit), we'll return None.
-        - Else if this is a new widget, it won't yet have a value and we'll
```

Comment:
> Else if this is a new widget, it won't yet have a value and we'll return None.

I didn't find the logic this was referring to, and it seems untrue, both before and after this PR, so I removed it.